src/Groups/types.jl

```diff
@@ -203,7 +209,29 @@ end
     AutomorphismGroupElem
 Element of a group of automorphisms.
 """
-const AutomorphismGroupElem = GAPGroupElem{AutomorphismGroup}
+const AutomorphismGroupElem = Union{GAPGroupElem{AutomorphismGroup},
+GAPGroupElem{AutomorphismGroup{PermGroup}},
+GAPGroupElem{AutomorphismGroup{PcGroup}},
+GAPGroupElem{AutomorphismGroup{FPGroup}},
+GAPGroupElem{AutomorphismGroup{MatrixGroup}}
```

Comment:
Ugh, this doesn't scale at all: it prevents GAP packages from adding further types in the future.

Reply:
If further types are added, we need to add rows here, of course. Or does a generalized procedure exists? (something like `GAPGroupElem{AutomorphismGroup{T}} where T <: GAPGroup` )

Reply:
The problem is that only we (as authors of `Oscar.jl`) can change this type, and also users only will get the change when they update to a new version. But people may want to add their own Julia packages extending this, and with the above setup, they cannot do that.

To resolve this, I think it would be best to take a step backwards and consider "which problem are we trying to solve here?" and "do we actually need to solve that problem?". Looking at the code, I think `AutomorphismGroupElem` is only used once in `master` (and twice in this branch):
```julia
function hom(x::AutomorphismGroupElem)
...
function isinner_automorphism(f::GAPGroupHomomorphism)
```
But in master we have this (which could and should be improved, e.g. to say `T <: GAPGroup`, but I think in general it works?):
```julia
function hom(x::GAPGroupElem{AutomorphismGroup{T}}) where T
```

So my solution would be to *remove* type `AutomorphismGroupElem` and adjust the 1-2 places using it, and any future places using it. If this ever starts to become a problem, we can revisit this.